Added the main body of code for Pachinko

Kortigus opened the pull request.

Kortigus committed:
Added the main body of code for Pachinko

ExtremeSkepticist commented on Pachinko/Pachinko.c:
```diff
@@ -0,0 +1,104 @@
+int main(void)
+{
+	int slot[3];
+	int credits = 1000;
```
Size_t instead of int

ExtremeSkepticist commented on Pachinko/Pachinko.c:
```diff
@@ -0,0 +1,104 @@
+{
+	int slot[3];
+	int credits = 1000;
+	int bet = 0;
```
Same as above

ExtremeSkepticist commented on Pachinko/Pachinko.c:
```diff
@@ -0,0 +1,104 @@
+
+	while (1)
+	{
+		if (credits < 100)
```
Use MINIMUM instead of 100

ExtremeSkepticist commented on Pachinko/Pachinko.c:
```diff
@@ -0,0 +1,104 @@
+		{
+			clearerr(stdin);
+			printf("Error reading line");
+			return 1;
```
If there's an error, and this returns zero, does a player keep playing?

ExtremeSkepticist commented on Pachinko/Pachinko.c:
```diff
@@ -0,0 +1,104 @@
+			printf("Error reading line");
+			return 1;
+		}
+		if (sscanf_s(line, "%d", bet) == 1 && *bet >= 100)
```
Use MINIMUM instead of 100

ExtremeSkepticist commented on Pachinko/Pachinko.c:
```diff
@@ -0,0 +1,104 @@
+	size_t i;
+	for (i = 0; i < 3; i++)
+	{
+		pickRandom(&slot[i]);
```
You don't need to make this into a function.

ExtremeSkepticist commented on Pachinko/Pachinko.c:
```diff
@@ -0,0 +1,104 @@
+/* Spins the slots */
+void playSlots(int slot[], size_t n)
+{
+	size_t i;
```
Erase this and use n

Kortigus replied:
should i erase i, or use it to iterate, and then replace the > 3 with > n

ExtremeSkepticist replied:
No thj sis a mistake. I was gonna say use n instead of 3.

ExtremeSkepticist commented on Pachinko/Pachinko.c:
```diff
@@ -0,0 +1,104 @@
+	{
+		switch (slot[i])
+		{
+		case 0: printf("%s", "@");
```
Use the following format
switch(slot[i])
{
case 0:
{
    printf("%s", "@");
    break;
}
case 1:
...
}

ExtremeSkepticist commented on Pachinko/Pachinko.c:
```diff
@@ -0,0 +1,104 @@
+				break;
+		case 3: printf("%s", "%");
+				break;
+		case 4: printf("%s", "7");
```
You can't have 4. (rand() % 4 generates 0 to 3)

ExtremeSkepticist commented on Pachinko/Pachinko.c:
```diff
@@ -0,0 +1,104 @@
+void payOut(int slot[], int* credits, int* bet)
+{
+	size_t i;
+	int* p = &slot[0];
```
Why do you need this?
Just use slot

Kortigus committed:
-Changed ints to size_t where ints are not needed
-Removed the pickRandom() as it was not needed
-Added functionality to check if the player attempts to bet more than they have.

ExtremeSkepticist commented on Pachinko/Pachinko.c:
```diff
@@ -5,68 +5,86 @@
 
 /* Gets the amount the player wants to bet and stores it into bet */
-int insertCoin(int* bet)
+insertCoin(size_t* bet, size_t* credits)
```
Where's the return type?

ExtremeSkepticist commented on Pachinko/Pachinko.c:
```diff
@@ -5,68 +5,86 @@
 		{
 			clearerr(stdin);
 			printf("Error reading line");
 			return 1;
```
If there's an error, you shouldn't let the roulette spin and let a user start over. That's what I meant

Kortigus committed:
-Fixed so the game exits if there is a read error when reading for a bet placement
